### ISSUE TITLE
Audio Pre-process Error

### ISSUE THREAD

**Issue by small-guang** [Fictionarry/ER-NeRF#10]
The following error occurs in DeepSpeec:
![image](https://github.com/Fictionarry/ER-NeRF/assets/57381375/2e2f7b79-ac52-4bcd-80af-3a1c91104f96)
The following error occurred while using HuBERT:
![image](https://github.com/Fictionarry/ER-NeRF/assets/57381375/06ddf8e3-99e4-4cdb-a7db-2821661d1bef)

**Comment by Fictionarry** [comment]
I guess it's caused by the undesired sampling rate of the input audio. You can use `librosa.resample` or other libs to convert the sampling rate to 16k.

**Comment by small-guang** [comment]
I use the following code to control the sample rate of 16k, but the above error still occurs. It seems that the sample rate is not the problem
```python
if samplerate != 16000:
    speech_16k = librosa.resample(speech_16k, samplerate, 16000)
```

**Comment by small-guang** [comment]
> I guess it's caused by the undesired sampling rate of the input audio. You can use `librosa.resample` or other libs to convert the sampling rate to 16k.
I use the following code to control the sample rate of 16k, but the above error still occurs. It seems that the sample rate is not the problem

```python
if samplerate != 16000:
    speech_16k = librosa.resample(speech_16k, samplerate, 16000)
```

**Comment by Fictionarry** [comment]
I'm sorry for that, however, the code itself is less likely to have bugs, since they are borrowed from existing successful projects and work well in my test in many cases. Maybe you could try more other .wav files and see what will happen.

**Comment by BingliangLi** [comment]
Using ffmpeg worked for me:
`ffmpeg -i input_audio_file.wav -ar 16000 output_audio_file.wav`

**Comment by MoroseYu** [comment]
Indeed, there is an issue with the audio, primarily related to the HuBERT preprocessing method. The same 16K WAV file works fine with the DeepSpeech method, but encounters errors with HuBERT. Even after processing the audio file using the command "ffmpeg -i input_audio_file.wav -ar 16000 output_audio_file.wav," the problem still persists with HuBERT. Please take this issue seriously and give it due attention.

**Comment by BingliangLi** [comment]
> Indeed, there is an issue with the audio, primarily related to the HuBERT preprocessing method. The same 16K WAV file works fine with the DeepSpeech method, but encounters errors with HuBERT. Even after processing the audio file using the command "ffmpeg -i input_audio_file.wav -ar 16000 output_audio_file.wav," the problem still persists with HuBERT. Please take this issue seriously and give it due attention.

Can confirm, DeepSpeech works fine with the command I provided above but not HuBERT.

**Comment by BingliangLi** [comment]
Potential solution:
In `data_utils/hubert.py` [line 80](https://github.com/Fictionarry/ER-NeRF/blob/main/data_utils/hubert.py#L80), change it to the following:
```
hubert_hidden = get_hubert_from_16k_speech(speech_16k)
# padding so can be reshaped
if hubert_hidden.shape[0] % 2 != 0:
    hubert_hidden = torch.nn.functional.pad(hubert_hidden, (0,0,0,1))
```
Seems be working for my data, anyone can confirm?
